libsolidity/ast/ASTAnnotations.h

```diff
@@ -162,6 +164,10 @@ struct ContractDefinitionAnnotation: TypeDeclarationAnnotation, StructurallyDocu
 	/// Mapping containing the nodes that define the arguments for base constructors.
 	/// These can either be inheritance specifiers or modifier invocations.
 	std::map<FunctionDefinition const*, ASTNode const*> baseConstructorArguments;
+	/// A graph with edges representing calls between functions that may happen during contract construction.
+	SetOnce<std::shared_ptr<CallGraph const>> creationCallGraph;
+	/// A graph with edges representing calls between functions that may happen in a deployed contract.
+	SetOnce<std::shared_ptr<CallGraph const>> deployedCallGraph;
```

Comment:
I'd prefer to use `CallGraph` as a value here but I had to use a pointer instead due to dependencies. `CallGraph.h` needs `AST.h` so including it here would create a dependency cycle. I had to use something that only requires a forward declaration.

And the reason that `CallGraph.h` can't just use `ASTForward.h` is that it uses `ASTNode::CompareByID` in member declarations.

Also, I wanted to at last have `unique_ptr` here but that does not work with an incomplete type so I had to use `shared_ptr` instead.